test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -145,19 +170,25 @@ def test_basic_save_and_load_state_dict(self, cpu_offload, fp16, state_dict_rank
         match as expected.
         """
         for model_call in [
+            partial(self._get_non_fsdp_root_module, cpu_offload=cpu_offload),
             partial(self._get_simple_nested_model, cpu_offload=cpu_offload),
             partial(self._get_simple_model, cpu_offload=cpu_offload),
         ]:
             model = model_call()
+            if self.rank == 0:
+                print(model)
             full_state_dict_mgr = self._get_full_state_dict_mgr(
                 model, state_dict_rank0_and_offload
             )
             with full_state_dict_mgr:
                 fsdp_state_dict = _get_state_dict(model, cpu_offload.offload_params, fp16)
 
-            self._validate_state_dict_contents(
-                fsdp_state_dict, state_dict_rank0_and_offload
-            )
+            # For non-FSDP roots, the non FSDP portion can still have parameters on rank 0,
```

Comment:
I may be missing some context. Can you clarify why the non-FSDP portion can _still_ have parameters on rank 0? (E.g. why not?)

I think maybe the intended comment was that for non-FSDP root models, the non-FSDP portion can still contribute parameters to the state dict even on _nonzero_ ranks?

```suggestion
            # For non-FSDP roots, the non FSDP portion can still contribute parameters to the state dict on nonzero ranks,
```

IIUC, this means that the check does not call `_validate_state_dict_contents()` for the `self._get_non_fsdp_root_module()` case. Will this be updated in the future?

Reply:
Yeah, miscommented. The check should be updated in the future ideally, I think I should probably just do it in this PR and get the refactoring done - good point.